monster-cards/src/actions/widgetStoreActions.ts

```diff
@@ -1,47 +1,37 @@
 import createAction from 'dojo-actions/createAction';
-import { assign } from 'dojo-core/lang';
 
 import { ChangeRecord, Card } from '../stores/cardStore';
 import widgetStore from '../stores/widgetStore';
 
-function createWidgetsRecords(item: Card) {
-	const cardId = item.id;
-	return Promise.all([
-		widgetStore.put(assign({}, <any> item, {
-			type: 'milestone-card',
-			cardId
-		})),
-		widgetStore.put(assign({}, <any> item, {
-			type: 'milestone-card-summary',
-			id: `summary-${item.id}`,
-			cardId
-		})),
-		widgetStore.put(assign({}, <any> item, {
-			type: 'card-description',
-			classes: [ 'animated', 'cardDetailsDescription' ],
-			id: `description-${item.id}`,
-			enterAnimation: 'fadeInRight',
-			exitAnimation: 'fadeOutLeft'
-		}))
-	]);
+import { CardState } from '../widgets/card/createCard';
+import { CardSummaryState } from '../widgets/card/createCardSummary';
+
+function getStates(cards: Card[]): [ CardState[], CardSummaryState[] ] {
+	const cardStates: CardState[] = [];
+	const cardSummaryStates: CardSummaryState[] = [];
+
+	cards.forEach(function (card) {
+		const {
```

Comment:
there were a lot more of them before i refactored, will put back on one line perhaps